src/App/Fossa/Analyze.hs

```diff
@@ -353,8 +365,9 @@ analyze cfg = Diag.context "fossa-analyze" $ do
   renderScanSummary (severity cfg) maybeEndpointAppVersion analysisResult $ Config.filterSet cfg
 
   -- Need to check if vendored is empty as well, even if its a boolean that vendoredDeps exist
-  let result = buildResult includeAll additionalSourceUnits filteredProjects firstPartyScanResults
-  case checkForEmptyUpload includeAll projectResults filteredProjects additionalSourceUnits firstPartyScanResults of
+  let licenseSourceUnits = mergeLicenseSourceUnits firstPartyScanResults $ lernieResultsSourceUnit =<< grepResults
```

Comment:
I need to return `LernieResults` rather than just a source unit so that I can use the search results from `LernieResults` to generate the scan summary (https://fossa.atlassian.net/browse/ANE-1164)

I may also need the warnings and errors while doing that, but I'm not sure yet.